src/stats/hexbin.jl

```diff
@@ -0,0 +1,136 @@
+"""
+    hexbin(xs, ys; kwargs...)
+
+Plots a heatmap with hexagonal bins for the observations `xs` and `ys`.
+
+## Attributes
+### Specific to `Hexbin`
+* `gridsize::Int = 20` sets the number of bins in x-direction
+* `mincnt::Int = 0` sets the minimal number of observations in the bin to be shown. If 0 all bins are shown, if 1 all with at least 1 observation.
+* `scale = identity` scales the number of data in the bins, eg. log10.
+### Generic
+* `colormap::Union{Symbol, Vector{<:Colorant}} = :viridis` sets the colormap that is sampled for numeric colors.
+* `colorrange::Tuple(<:Real,<:Real} = Makie.automatic`  sets the values representing the start and end points of `colormap`.
+"""
+@recipe(Hexbin) do scene
+    Attributes(;
+    colormap = theme(scene, :colormap),
+    colorrange=Makie.automatic,
+    gridsize = 20,
+    mincnt = 0,
+    scale=identity)
+end
+
+function Makie.plot!(hb::Hexbin{<:Tuple{<:AbstractVector{<:Any}, <:AbstractVector{<:Any}}})
+    x,y =hb[1:2]
+   
+    polys = Observable(Polygon[])
```

Comment:
Polygon is an abstract type so this doesn't really help in terms of type stability.

Reply:
you could construct a fake poly and do `typeof(fakepoly)[]` if the type is hard to write manually (sometimes the case with GeometryBasics types I think)